AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js

```diff
@@ -99,11 +119,71 @@ define([
          * Selects displayed image in media gallery
          */
         selectDisplayedImageInMediaGallery: function () {
+            if (!this.isMediaBrowser()) {
+                this.selectDisplayedImageForNewMediaGallery()
+            } else {
+                this.selectDisplayedImageForOldMediaGallery();
+            }
+        },
+
+        /**
+         * Selects displayed image in media gallery for old gallery
+         */
+        selectDisplayedImageForOldMediaGallery: function () {
             var image = mediaGallery.locate(this.preview().displayedRecord().path);
 
             image ? image.click() : mediaGallery.notLocated();
         },
 
+        /**
+         * Selects displayed image in media gallery for new gallery
+         */
+        selectDisplayedImageForNewMediaGallery: function () {
+            var self = this,
+                imagePath = self.preview().displayedRecord().path,
+                imageFolders = mediaGallery.getImageFolders(imagePath),
+                imageFilename = imagePath.substring(imagePath.lastIndexOf('/') + 1),
+                locatedImage = $('div[data-row="file"]:has(img[alt=\"' + imageFilename + '\"])'),
+                image,
+                subscription;
+
+            if (!locatedImage.length) {
+                subscription = this.imageItems.subscribe(function () {
+                    locatedImage = $('div[data-row="file"]:has(img[alt=\"' + imageFilename + '\"])');
+                    image = locatedImage.length ? locatedImage : false;
+
+                    if (!image) {
+                        mediaGallery.notLocated();
+
+                        return;
+                    }
+
+                    self.selectRecord(image);
+                    subscription.dispose();
+                });
+            }
+
+            if (imageFolders.length) {
+                imageFolders[0].click();
```

Comment:
If the directory with the image was selected before the locate action was called - the selection of the directory is dropped after this click.

Is it possible to use `MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js.setActiveNodeFilter` method call to select the required directory by path? (even without searching for the folder beforehand)

Reply:
Fixed and I added small change in the `setActiveNodeFilter` function, please check. 
Thank you!